common/djangoapps/edxmako/middleware.py

```diff
@@ -62,4 +70,6 @@ def get_template_request_context():
     for processor in get_template_context_processors():
         context.update(processor(request))
 
+    request_cache_dict[cache_key] = context
```

Comment:
I added the mako caching just because it seemed like the easiest-to-fix thing to offset the overhead added by capa inlining. The middleware just attaches a request to a var so that it can send it to the context processors -- it doesn't cache the results of the context processors.

Though honestly, I'm not totally sure what's going on in there either. I'll dig deeper into that.